pyteal/ast/subroutine.py

```diff
@@ -610,12 +611,77 @@ def method_signature(self, overriding_name: str = None) -> str:
         return f"{overriding_name}({','.join(args)}){self.type_of()}"
 
     def method_spec(self) -> sdk_abi.Method:
-        skip_names = ["return", "output"]
+        skip_names = ["return", self.OUTPUT_ARG_NAME]
+
+        desc: str = ""
+        arg_descs: dict[str, str] = {}
+        output_desc: str = ""
+        return_desc: str = ""
+
+        def get_method_desc(short_desc: Optional[str], long_desc: Optional[str]) -> str:
+            """
+            method description is based on short desc and long desc, with 4 cases of outputs:
+            - if only short (or long) desc, then output only short (or long) desc
+            - if neither exists, then outputs empty string
+            - if both exists, concat with break line
+            """
+            if not short_desc:
+                return long_desc if long_desc else ""
+            else:
+                _method_desc = short_desc
+                if long_desc:
+                    _method_desc = _method_desc + "\n" + long_desc
+                return _method_desc
+
+        def get_ret_desc(_ret_desc: str, _output_desc: str) -> str:
+            """
+            return description is based on return desc and output desc:
+            - if only return (or output) desc, then output only return (or output) desc
+            - if neither exists, then outputs empty string
+            - if both exists, concat with space
+            """
+            if not _ret_desc:
+                return _output_desc
+            else:
+                res = _ret_desc
+                if _output_desc:
+                    res = res + " " + _output_desc
+                return res
+
+        if self.subroutine.implementation.__doc__:
+            docstring = parse_docstring(self.subroutine.implementation.__doc__)
+            method_desc = get_method_desc(
+                docstring.short_description, docstring.long_description
+            )
+
+            desc = " ".join(
+                [
+                    i.strip()
+                    for i in method_desc.split("\n")
+                    if not (i.isspace() or len(i) == 0)
+                ]
+            )
```

Comment:
This does not look like it will satisfy the first item from #516.

To be sure, the tests should include multiple "paragraphs" in their docstrings. Right now they only populate the short description.